src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,62 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.util.*;
 
 public class TournamentStats {
 
+    private static final String STOP = "STOP";
+    private static final int FIRST_NAME = 1;
+    private static final int LAST_NAME = 2;
+    private static final int RESULT = 3;
+
+    private static final int ASCENDING = 1;
+    private static final int DESCENDING = 2;
+
     void run(Scanner scanner) {
-        // tutaj dodaj swoje rozwiązanie
-        // użyj przekazanego scannera do wczytywania wartości
+        String result;
+        List<Participant> participants = new ArrayList<>();
+        do {
+            result = getResult(scanner);
+            addParticipantResult(result, participants);
+        } while (!result.equals(STOP));
+        Comparator<Participant> sortComparator = createComparator(scanner);
+        participants.sort(sortComparator);
+        TournamentStatsWriter writer = new TournamentStatsWriter();
+        writer.writeStatsToFile(participants);
+    }
+
+    private Comparator<Participant> createComparator(Scanner scanner) {
+        Comparator<Participant> sortComparator = null;
+        System.out.println("Po jakim parametrze posortować? (" + FIRST_NAME + " - imię, "
+                + LAST_NAME + " - nazwisko, " + RESULT + " - wynik)");
+        int option = scanner.nextInt();
+        System.out.println("Sortować rosnąco czy malejąco? (" + ASCENDING + " - rosnąco, " + DESCENDING + " - malejąco)");
+        int option2 = scanner.nextInt();
+        if (option == FIRST_NAME && option2 == ASCENDING) {
+            sortComparator = new FirstNameComparator();
+        } else if (option == FIRST_NAME && option2 == DESCENDING) {
+            sortComparator = new FirstNameComparator().reversed();
+        } else if (option == LAST_NAME && option2 == ASCENDING) {
+            sortComparator = new LastNameComparator();
+        } else if (option == LAST_NAME && option2 == DESCENDING) {
+            sortComparator = new LastNameComparator().reversed();
+        } else if (option == RESULT && option2 == ASCENDING) {
+            sortComparator = new ResultComparator();
+        } else if (option == RESULT && option2 == DESCENDING) {
+            sortComparator = new ResultComparator().reversed();
+        }
+        return sortComparator;
```

Comment:
Zamisat tylu koniunkcji, lepiej zapisać dwa osobne ify, pierwszy odpowiedzialny za utworzenie komparatora, drugi za jego ewentualne odwrócenie. Jeżeli kryteriów sortowania będzie więcej, to zamiast za każdym razem dodawać dwa if-elseif, dodawany będzie tylko jeden